libuavcan/src/transport/uc_can_acceptance_filter_configurator.cpp

```diff
@@ -115,12 +103,27 @@ int16_t CanAcceptanceFilterConfigurator::computeConfiguration()
 int16_t CanAcceptanceFilterConfigurator::applyConfiguration(void)
 {
     CanFilterConfig filter_conf_array[MaxCanAcceptanceFilters];
-    const unsigned int filter_array_size = multiset_configs_.getSize();
+    unsigned int filter_array_size = multiset_configs_.getSize();
+
+    const uint16_t acceptance_filters_number = getNumFilters();
+    if (acceptance_filters_number == 0)
+    {
+        UAVCAN_TRACE("CanAcceptanceFilter", "No HW filters available");
+        return -ErrDriver;
+    }
+
+    if (filter_array_size > acceptance_filters_number)
+    {
+        UAVCAN_TRACE("CanAcceptanceFilter", "Too many filter configurations. Executing computeConfiguration()");
+        computeConfiguration(IgnoreAnonymousMessages);
```

Comment:
If I understand the logic correctly, this line is supposed to call `mergeConfigurations()`, not `computeConfiguration()`.